### PR TITLE
Bye Bye chunkperf.php :wave: 

### PR DESCRIPTION
**Pull request by kesselb**
I found this gem in our psalm-baseline.xml today :rofl: 

It's a script to upload a file in chunks to Nextcloud. 
~Fixed the psalm warning, the autoloader import and added a usage message.~
Decided to remove the script.

```
php chunkperf.php http://server:80/index.php admin password /home/daniel/Downloads/herbst_version2.mp4 10

MKCOL http://server:80/index.php/uploads/admin/transfer6282b1818dce32.43375928 ... 405 - 0.21005487442017 seconds

PUT http://server:80/index.php/uploads/admin/transfer6282b1818dce32.43375928/0 ... 405 - 0.21709990501404 seconds

PUT http://server:80/index.php/uploads/admin/transfer6282b1818dce32.43375928/1 ... 405 - 0.21805095672607 seconds

PUT http://server:80/index.php/uploads/admin/transfer6282b1818dce32.43375928/2 ... 405 - 0.24585294723511 seconds

PUT http://server:80/index.php/uploads/admin/transfer6282b1818dce32.43375928/3 ... 405 - 0.24216413497925 seconds

PUT http://server:80/index.php/uploads/admin/transfer6282b1818dce32.43375928/4 ... 405 - 0.22598385810852 seconds

MOVE http://server:80/index.php/uploads/admin/transfer6282b1818dce32.43375928/.file ... 405 - 0.20804595947266 seconds

```

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]

**Comment by kesselb** [comment]
image

I don't know how to fix this error :confused:

File: apps/dav/bin/chunkperf.php [hunk 1]
```diff
@@ -20,25 +20,31 @@
...
- * @return mixed
+ * @param Client $client
+ * @param string $method
+ * @param string $uploadUrl
+ * @param string|resource|null $data
+ * @param array $headers
+ * @return array
  */
-function request($client, $method, $uploadUrl, $data = null, $headers = []) {
+function request($client, $method, $uploadUrl, $data = null, $headers = []): array {
```
Comment (come-nc) [inline apps/dav/bin/chunkperf.php:41]:
```suggestion
function request(Client $client, string $method, string $uploadUrl, $data = null, array $headers = []): array {
```

**Review by come-nc — APPROVED** [review]
Should be an occ command no?

**Comment by kesselb** [comment]
> Should be an occ command no?

Maybe ;) It seems the script was added with https://github.com/nextcloud/server/commit/72f5c539e887d2671641222d33848e28a936d771 as helper to test and benchmark chunked uploads. It's broken for quite some time (because the path to 3rdparty autoloader was wrong). I have really no idea about the psalm warnings. Maybe it's easier to just remove it :thinking:

**Comment by CarlSchwan** [comment]
I would remove it completely (see https://github.com/nextcloud/server/pull/31943#issuecomment-1119087854)

**Review by MichaIng — APPROVED** [review]

**Review by CarlSchwan — APPROVED** [review]

**Pull request merged**